lib/pre_build.ts

```diff
@@ -322,26 +328,33 @@ export function isInstantiated() {
   return instanceWithExports != null;
 }
 
-async function instantiateModule(transform) {
-  switch (wasm_url.protocol) {
+/**
+ * @param {InstantiateOptions} opts
+ */
+async function instantiateModule(opts) {
+  const wasmUrl = opts.url ?? new URL("${
+    getWasmFileNameFromCrate(crate)
+  }", import.meta.url);
+  const decompress = opts.decompress;
+  switch (wasmUrl.protocol) {
     case "file:": {
       if (typeof Deno !== "object") {
         throw new Error("file urls are not supported in this environment");
       }
 
-      if ("permissions" in Deno) await Deno.permissions.request({ name: "read", path: wasm_url });
-      const wasmCode = await Deno.readFile(wasm_url);
-      return WebAssembly.instantiate(!transform ? wasmCode : transform(wasmCode), imports);
+      if ("permissions" in Deno) await Deno.permissions.request({ name: "read", path: wasmUrl });
+      const wasmCode = await Deno.readFile(wasmUrl);
+      return WebAssembly.instantiate(!decompress ? wasmCode : decompress(wasmCode), imports);
```

Comment:
Drive by review: We should use the `fetch` path for local files too. That way we can use the streaming wasm optimization.

Reply:
Thanks! Updated.